test/functional/ui/output_spec.lua

```diff
@@ -39,4 +40,17 @@ describe("shell command :!", function()
       {3:-- TERMINAL --}                                    |
     ]])
   end)
+
+  it("throttles output >~20KB", function()
+    child_session.feed_data(":!for i in $(seq 2 1800); do echo XXXXXXXXXX; done\n")
+    screen:expect([[
+      XXXXXXXXXX                                        |
+      XXXXXXXXXX                                        |
+      XXXXXXXXXX                                        |
+      XXXXXXXXXX                                        |
+      ..                                                |
+      {10:Press ENTER or type command to continue}{1: }          |
+      {3:-- TERMINAL --}                                    |
```

Comment:
This test is a bit fragile, I'm not sure only two dots will appear on all platforms, it depends on the buffer size.

If this is too fragile, I may need to add a function to `screen.lua` that checks for a pattern without asserting the exact screen state.